### PR TITLE
build: enable C11 globally and C11 atomics on MSVC

### PR DESCRIPTION
**Pull request by compnerd**
This accounts for local changes to the repository that are unguarded and undocumented.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please smoke test

File: CMakeLists.txt
Comment (d-ronnqvist) [inline CMakeLists.txt:1]: Just to double check my understanding; this only affects CMake builds, correct?

Comment (matthewbastien) [reply CMakeLists.txt:1]: Yes, this change only affects CMake builds.

**Review by d-ronnqvist — APPROVED** [review]

File: CMakeLists.txt
Comment (compnerd) [reply CMakeLists.txt:1]: This change only impacts the CMake build, though the underlying issue impacts all platforms and should do something similar on the SPM side as well.

**Pull request merged**